<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindowClass</class>
 <widget class="QMainWindow" name="MainWindowClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>811</width>
    <height>704</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ApiUsageDemo</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QFrame" name="frame_2">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QToolButton" name="toolButtonEds">
         <property name="text">
          <string>EDS</string>
         </property>
         <property name="icon">
          <iconset resource="cloudconsole.qrc">
           <normaloff>:/ApiUsageDemo/Images/Icons/EdsLogo.png</normaloff>:/ApiUsageDemo/Images/Icons/EdsLogo.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>64</width>
           <height>64</height>
          </size>
         </property>
         <property name="shortcut">
          <string>Ctrl+1</string>
         </property>
         <property name="toolButtonStyle">
          <enum>Qt::ToolButtonTextUnderIcon</enum>
         </property>
         <property name="autoRaise">
          <bool>false</bool>
         </property>
         <property name="arrowType">
          <enum>Qt::NoArrow</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="toolButtonMws">
         <property name="text">
          <string>MWS</string>
         </property>
         <property name="icon">
          <iconset resource="cloudconsole.qrc">
           <normaloff>:/ApiUsageDemo/Images/Icons/MwsLogo.png</normaloff>:/ApiUsageDemo/Images/Icons/MwsLogo.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>64</width>
           <height>64</height>
          </size>
         </property>
         <property name="shortcut">
          <string>Ctrl+2</string>
         </property>
         <property name="toolButtonStyle">
          <enum>Qt::ToolButtonTextUnderIcon</enum>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>589</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="page">
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QLabel" name="labelEnginioDataStorage">
          <property name="text">
           <string>Enginio Data Storage</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="2">
           <widget class="QLineEdit" name="lineEditEdsBackendId">
            <property name="text">
             <string>5379dea0698b3c1dc00cdf57</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="labelEdsBackendId">
            <property name="text">
             <string>EDS_BACKEND_ID</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QLineEdit" name="lineEditEdsinstanceAddress">
            <property name="text">
             <string>https://api.engin.io</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="labelEdsInstanceAddress">
            <property name="text">
             <string>EDS_INSTANCE_ADDRESS</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="labelCollection">
            <property name="text">
             <string>Collection</string>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QLineEdit" name="lineEditCollection">
            <property name="text">
             <string>tasks</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QTabWidget" name="tabWidgetObjectOperations">
            <property name="currentIndex">
             <number>0</number>
            </property>
            <widget class="QWidget" name="tabFind">
             <attribute name="title">
              <string>find</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_8">
              <item>
               <widget class="QFrame" name="frame">
                <property name="frameShape">
                 <enum>QFrame::StyledPanel</enum>
                </property>
                <property name="frameShadow">
                 <enum>QFrame::Raised</enum>
                </property>
                <layout class="QGridLayout" name="gridLayout_2">
                 <item row="0" column="0">
                  <widget class="QLabel" name="label_5">
                   <property name="text">
                    <string>Limit</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="1">
                  <widget class="QSpinBox" name="spinBoxFindLimit">
                   <property name="maximum">
                    <number>1000</number>
                   </property>
                   <property name="value">
                    <number>100</number>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="0">
                  <widget class="QLabel" name="label_9">
                   <property name="text">
                    <string>Offset</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="1">
                  <widget class="QSpinBox" name="spinBoxFindOffset">
                   <property name="maximum">
                    <number>1000</number>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_4">
                <property name="text">
                 <string>Query Object (Optional)</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPlainTextEdit" name="plainTextEditFindInput">
                <property name="plainText">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tabFindOne">
             <attribute name="title">
              <string>findOne</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_7">
              <item>
               <widget class="QLabel" name="label_8">
                <property name="text">
                 <string>ObjectId</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="lineEditFindOneObjectId"/>
              </item>
              <item>
               <spacer name="verticalSpacer_3">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>395</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tabInsert">
             <attribute name="title">
              <string>insert</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <item>
               <widget class="QLabel" name="label">
                <property name="text">
                 <string>JSON Object to be inserted</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPlainTextEdit" name="plainTextEditInsertInput">
                <property name="plainText">
                 <string>{
   &quot;name&quot;: &quot;John Doe&quot;,
   &quot;age&quot;: 42
}
</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tabUpdate">
             <attribute name="title">
              <string>update</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_6">
              <item>
               <widget class="QLabel" name="label_3">
                <property name="text">
                 <string>ObjectId</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="lineEditUpdateObjectId"/>
              </item>
              <item>
               <widget class="QLabel" name="label_7">
                <property name="text">
                 <string>JSON Object (fields to be updated).</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPlainTextEdit" name="plainTextEditUpdateInput">
                <property name="plainText">
                 <string>{
   &quot;age&quot;: 24
}
</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tabRemove">
             <attribute name="title">
              <string>remove</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_5">
              <item>
               <widget class="QLabel" name="label_6">
                <property name="text">
                 <string>ObjectId</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="lineEditRemoveObjectId"/>
              </item>
              <item>
               <spacer name="verticalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>395</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>Results</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPlainTextEdit" name="plainTextEditOutput"/>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="pushButtonExecute">
            <property name="text">
             <string>Execute</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_2">
       <layout class="QVBoxLayout" name="verticalLayout_10">
        <item>
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>Managed WebSocket</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <widget class="QLabel" name="labelMwsGatewayId">
            <property name="text">
             <string>MWS_GATEWAY_ID</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLineEdit" name="lineEditMwsGateWayId">
            <property name="text">
             <string>5386110b1ba00f02f2000002</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QLineEdit" name="lineEditMwsInstanceAddress">
            <property name="text">
             <string>https://mws-eu-1.qtc.io</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="labelMwsInstanceAddress">
            <property name="text">
             <string>MWS_INSTANCE_ADDRESS</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QPlainTextEdit" name="plainTextEditMwsInput"/>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_9">
            <item>
             <widget class="QPushButton" name="pushButtonMwsConnectOrDisconnect">
              <property name="text">
               <string>Connect</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer_4">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="pushButtonMwsExecute">
              <property name="text">
               <string>Execute</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTableWidget" name="tableWidgetMwsOutput"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>811</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="action_Quit"/>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Help"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="action_Quit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="cloudconsole.qrc"/>
  <include location="cloudconsole.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButtonMwsConnectOrDisconnect</sender>
   <signal>clicked()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>doMwsConnect()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>750</x>
     <y>158</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonExecute</sender>
   <signal>clicked()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>doEdsExecute()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>755</x>
     <y>644</y>
    </hint>
    <hint type="destinationlabel">
     <x>412</x>
     <y>353</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButtonEds</sender>
   <signal>clicked()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>doSwitchToEds()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>54</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>405</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButtonMws</sender>
   <signal>clicked()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>doSwitchToMws()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>54</x>
     <y>188</y>
    </hint>
    <hint type="destinationlabel">
     <x>405</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonMwsExecute</sender>
   <signal>clicked()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>doMwsExecute()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>750</x>
     <y>386</y>
    </hint>
    <hint type="destinationlabel">
     <x>405</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Quit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>405</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>doEdsExecute()</slot>
  <slot>doMwsConnect()</slot>
  <slot>doMwsExecute()</slot>
  <slot>doSwitchToEds()</slot>
  <slot>doSwitchToMws()</slot>
 </slots>
</ui>
